apps/omg_watcher/lib/exit_validator/validator.ex

```diff
@@ -14,24 +14,65 @@
 
 defmodule OMG.Watcher.ExitValidator.Validator do
   @moduledoc """
-  Fragment of imperative shell for ExitValidator. Validates exits.
+    Contains 'challenge_fastly_invalid_exits' and 'challenge_slowly_invalid_exits' functions
+    resposible for validating exits.
+
+    The exit validators covers following cases:
```

Comment:
ok so this way, lots of context is missed as is the rationale and explanations "why like this". Also the "next generation exit validation" is lost and we should leave a trace behind after our discussions on the "new approach" (aka "spend on `finalizeExit`").

To fix that I propose:
1/ in this module, drop point `c) (BG)...`
2/ in this module provide some context: what is `FV`, `SV` `M_SV` and explain how the exit validation works (imagine you really don't know, and want to find out). E.g. what does "Causes `:invalid_exit`" exactly mean (an event is emitted)
3/ add a `exit_validation.md` file in `docs/`, that contains the full description and rationale that we've arrived at, so what I posted on slack + anything else required or that we added/clarified. It is in that file, where you would **include** the explanation of the role of `(BG) - BlockGetter` in handling invalid exits
4/ link to the `.md` file from this `moduledoc`, saying that this doc provides more detail and rationale

Reply:
I created docs/exit_validation.md + in the `OMG.Watcher.ExitValidator.Validator` I left follwoing comments:
```
@moduledoc """
    Contains 'challenge_fastly_invalid_exits' and 'challenge_slowly_invalid_exits' functions
    used for current exit validation design which consits of `FastValidator` and `SlowValidator`.

    See docs/exit_validation.md for more information.
  """

@doc """
    Validates exits and pushes them to `DB.EthEvent`. if exit is invalid then emits `:invalid_exit` event
  """

  @doc """
    Validates and spends exits in the "OMG.API.State" and if exit is invalid then emits `:invalid_exit` event
  """
```